documentation/api/development.md

```diff
@@ -0,0 +1,72 @@
+3. Run `make setup-local` to install dependencies
+4. Run `make init start` to build the image and start the container.
+5. Navigate to `localhost:8080/docs` to access the Swagger UI.
+6. Run `make run-logs` to see the logs of the running API container
+7. Run `make stop` when you are done to delete the container.
+
+## Some Useful Commands
+
+`make test` will run all of the tests. Additional arguments can be passed to this command which will be passed to pytest like so: `make test args="tests/api/route -v"` which would run all tests in the route folder with verbosity increased. See the [Pytest Docs](https://docs.pytest.org/en/7.1.x/reference/reference.html#command-line-flags) for more details on CLI flags you can set.
+
+`make clean-volumes` will spin down the docker containers + delete the volumes. This can be useful to reset your DB, or fix any bad states your local environment may have gotten into.
+
+See the [Makefile](/api/Makefile) for a full list of commands you can run.
+
+## Docker and Native Development
+
+Several components like tests, linting, and scripts can be run either inside of the Docker container, or outside on your native machine.
+Running in Docker is the default, but on some machines like the M1 Mac, running natively may be desirable for performance reasons.
+
+You can switch which way many of these components are run by setting the `PY_RUN_APPROACH` env variable in your terminal.
+
+* `export PY_RUN_APPROACH=local` will run these components natively
+* `export PY_RUN_APPROACH=docker` will run these within Docker
+
+Note that even with the native mode, many components like the DB and API will only ever run in Docker, and you should always make sure that any implementations work within docker.
+
+Running in the native/local approach may require additional packages to be installed on your machine to get working.
+
+## Environment Variables
+
+Most configuration options are managed by environment variables.
+
+Environment variables for local development are stored in the [local.env](/api/local.env) file. This file is automatically loaded when running. If running within Docker, this file is specified as an `env_file` in the [docker-compose](/docker-compose.yml) file, and loaded [by a script](/api/src/util/local.py) automatically when running most other components outside the container.
+
+Any environment variables specified directly in the [docker-compose](/docker-compose.yml) file will take precedent over those specified in the [local.env](/api/local.env) file.
+
+## Authentication
+
+This API uses a very simple [ApiKey authentication approach](https://apiflask.com/authentication/#use-external-authentication-library) which requires the caller to provide a static key. This is specified with the `API_AUTH_TOKEN` environment variable.
+
+## Next steps
+
+Now that you're up and running, read the [application docs](README.md) to familiarize yourself with the application.
```

Comment:
This links to the readme in the documentation folder, which in turn links to the readme in the api folder. Can we just link directly to the api folder?

Reply:
I made this change, thanks.